test/deriving_warning/dune

```diff
@@ -1,4 +1,4 @@
 (library
  (name ppx_warning_flags_testing_on_tests)
  (preprocess
-  (pps ppx_warning_flags_testing)))
+  (pps ppx_warning_flags_testing -deriving-allow-unused-code-warnings)))
```

Comment:
To also add a test for this new cmd-line flag, we could have two tests, one with this flag and one without.

Reply:
Done.